compiler/qsc_frontend/src/compile/preprocess.rs

```diff
@@ -193,3 +197,75 @@ fn matches_config(attrs: &[Box<Attr>], capabilities: TargetCapabilityFlags) -> b
     capabilities.contains(found_capabilities)
         && (disallowed_capabilities.is_empty() || !capabilities.contains(disallowed_capabilities))
 }
+
+// Visitor to remove spans from circuit callables defined in QSC files.
+// This will remove the spans for the contents of these circuit callables,
+// but it is important for the language server that the span for the callable itself
+// is preserved, so that the user can navigate to the definition of the callable.
+pub(crate) struct RemoveCircuitSpans {
+    qsc_spans: Vec<Span>,
+    inside_stmt: bool,
+}
+
+impl RemoveCircuitSpans {
+    pub(crate) fn new(sources: &SourceMap) -> Self {
+        let qsc_spans = sources
+            .iter()
+            .filter(|source| {
+                std::path::Path::new(source.name.as_ref())
+                    .extension()
+                    .is_some_and(|ext| ext.eq_ignore_ascii_case("qsc"))
+            })
+            .map(|source| {
+                let start = source.offset;
+                let end = start
+                    + u32::try_from(source.contents.len()).expect("source length exceeds u32::MAX");
+                Span { lo: start, hi: end }
+            })
+            .collect();
+
+        Self {
+            qsc_spans,
+            inside_stmt: false,
+        }
+    }
+}
+
+impl MutVisitor for RemoveCircuitSpans {
+    fn visit_package(&mut self, package: &mut Package) {
+        // We only want to visit namespaces
+        package.nodes.iter_mut().for_each(|n| match n {
+            TopLevelNode::Namespace(ns) => self.visit_namespace(ns),
+            TopLevelNode::Stmt(_) => {}
+        });
+    }
+
+    fn visit_namespace(&mut self, namespace: &mut Namespace) {
+        // We only want to visit circuit callables
+        namespace.items.iter_mut().for_each(|item| {
+            if let ItemKind::Callable(callable) = item.kind.as_mut() {
+                // Check if the callable's span is inside a QSC file's span
+                self.qsc_spans
+                    .iter()
+                    .any(|s| s.lo <= callable.span.lo && s.hi >= callable.span.hi)
+                    .then(|| {
+                        self.visit_callable_decl(callable);
+                    });
+            }
+        });
+    }
+
+    fn visit_stmt(&mut self, stmt: &mut Stmt) {
+        self.inside_stmt = true;
+        walk_stmt(self, stmt);
+        self.inside_stmt = false;
+    }
```

Comment:
```suggestion
    fn visit_stmt(&mut self, stmt: &mut Stmt) {
        stmt.span.lo = 0;
        stmt.span.hi = 0;
        walk_stmt(self, stmt);
    }
```

Reply:
Is it safe enough for the debugger to remove spans only from the statement node but keep the spans for the underlying expressions, or are there circumstances where we want to remove the spans of the underlying expressions also?

Reply:
Went with only de-spanning statements as described here. It makes for simpler, easier to maintain code. One difference from the suggested code, however, is that I used the `Span::default()` instead of setting 0. This is because the debugger logic to check if it should skip a statement checks against `Span::default()`, so it is safer to be consistent with that.